java/src/main/java/com/powsybl/dataframe/update/UpdatingDataframe.java

```diff
@@ -0,0 +1,48 @@
+package com.powsybl.dataframe.update;
+
+import org.apache.commons.lang3.tuple.Pair;
+import java.util.List;
+import java.util.Set;
+
+public interface UpdatingDataframe {
+
+    void addIndex(String index);
```

Comment:
The best design would be not to have the setters/adders methods in the interface, we only need the getters which will be used by the mapper.
The setters/adders need only to be known at build time, where we know the actual implementation class.